it/pages/contactPreference/ContactPreferenceConfirmationPageSpec.scala

```diff
@@ -16,44 +16,71 @@
 
 package pages.contactPreference
 
-import models.contactPreferences.ContactPreference.paper
+import models.contactPreferences.ContactPreference._
 import pages.BasePageISpec
 import play.api.http.Status
 import play.api.libs.ws.WSResponse
+import stubs.VatSubscriptionStub
 
 class ContactPreferenceConfirmationPageSpec extends BasePageISpec {
 
-  val path = "/confirmation-email-preference"
+  val emailPath = "/confirmation-email-preference"
+  val letterPath = "/confirmation-letter-preference"
 
-  "Calling the EmailPreferenceConfirmation .show method" when {
-
-    def show: WSResponse = get(
-      path,
-      formatValidationEmail(Some("asd@asd.com")) ++
-        formatEmailPrefUpdate(Some("true")) ++
-        formatEmailPrefConfirmation(Some("true")) ++
-        formatLetterToEmailPrefConfirmation(Some("true")) ++
-        formatCurrentContactPref(Some(paper))
-    )
+  "Calling GET /confirmation-:changeType-preference" when {
 
     "the user is authenticated" when {
 
-      "the data is valid" should {
+      "changeType is email" when {
+
+        "session data is valid" should {
 
-        "display the page correctly" in {
+          "display the page correctly" in {
 
-          given.user.isAuthenticated
+            given.user.isAuthenticated
 
-          When("the emailPreferenceConfirmation page is called")
-          val result = show
+            When("GET /confirmation-email-preference is called")
+            val result = get(
+              emailPath,
+              formatCurrentContactPref(Some(paper)) ++
+              formatValidationEmail(Some("asd@asd.com")) ++
+              formatLetterToEmailPrefConfirmation(Some("true"))
+            )
 
-          result should have(
-            httpStatus(Status.OK),
-            pageTitle(generateDocumentTitle("contactPrefConfirmation.title"))
-          )
+            result should have(
+              httpStatus(Status.OK),
+              pageTitle(generateDocumentTitle("contactPrefConfirmation.title"))
+            )
+          }
         }
       }
 
+      "changeType is letter" when {
+
+        "session data is valid" should {
+
+          "display the page correctly" in {
+
+            given.user.isAuthenticated
+
+            When("GET /confirmation-letter-preference is called")
+
+            And("Current address is retrieved")
+            VatSubscriptionStub.stubCustomerInfo
+
+            val result = get(
+              letterPath,
+              formatCurrentContactPref(Some(digital)) ++
+              formatEmailToLetterPrefConfirmation(Some("true"))
+            )
+
+            result should have(
+              httpStatus(Status.OK),
+              pageTitle(generateDocumentTitle("contactPrefConfirmation.title"))
+            )
+          }
+        }
+      }
     }
 
     "a user is an authenticated agent" should {
```

Comment:
This test description says it will "render the Agent unauthorised page" when it's just simply redirecting to the client VAT number page. Also the `EmailToUse` page is incorrectly being referenced both here and in the test below.